### ISSUE TITLE
Any way to speed up builds on Windows?

### ISSUE THREAD

**Issue by pkgw** [mcgoo/vcpkg-rs#25]
For Tectonic, the Windows vcpkg builds are now timing out about 75% of the time because of the 60-minute job time limit for the Azure Pipeline build agents that we use. The lion's share of that time is taken up by the `cargo vcpkg build` of Tectonic's dependencies. I don't suppose there are any options that we could add/activate that would reduce how long things take? In particular, in the past I've noticed that vcpkg sometime builds both debug and release versions of its outputs ... does that happen to be the case with `cargo vcpkg build`?

**Comment by pkgw** [comment]
As it happens, I just got some evidence that maybe `cargo vcpkg build` *is* creating both debug and release artifacts. For Tectonic I've been testing a workaround where I do the `cargo vcpkg build` operations in a different job than the main build, and I ran into an error while trying to delete the ICU temporaries:

```
+ rm -rf target/vcpkg/buildtrees target/vcpkg/downloads
rm: cannot remove 'target/vcpkg/buildtrees/icu/x64-windows-static-dbg/common': Directory not empty
rm: cannot remove 'target/vcpkg/buildtrees/icu/x64-windows-static-dbg/data': Directory not empty
[...]
rm: cannot remove 'target/vcpkg/buildtrees/icu/x64-windows-static-rel/common': Directory not empty
rm: cannot remove 'target/vcpkg/buildtrees/icu/x64-windows-static-rel/data': Directory not empty
[...]
```

(I'm not sure why I can't `rm -rf` the ICU build tree, but that's a separate issue. Files might be getting marked read-only or something?)

**Comment by pkgw** [comment]
Looks like this is surprisingly poorly supported in vcpkg?

https://github.com/microsoft/vcpkg/issues/143

**Comment by mcgoo** [comment]
The vcpkg (and cargo vcpkg) default is to build both debug and release as you say. I don't know how to link Rust code to the debug build, so it has always been extra work for nothing. I missed the moment to make it the default for cargo vcpkg, but I'd like it to be available at the very least.

It's not that hard to request it - cargo-vcpkg would just have to add `set(VCPKG_BUILD_TYPE release)` to the triplet file on the fly. It will make the vcpkg tree dirty all the time so changing the rev in Cargo.toml will need a stash-pull-pop shuffle I suppose.

Apparently there has been some work done on [caching](https://devblogs.microsoft.com/cppblog/vcpkg-accelerate-your-team-development-environment-with-binary-caching-and-manifests/) but I have not tried it.

**Comment by pkgw** [comment]
Or is there a convenient way to create and use a custom triplet file that's kept outside of the vcpkg repo, maybe?

**Comment by mcgoo** [comment]
I'm still thinking about what the best way to add this capability is, but I think it's just a matter of adding a flag to the metadata.

If you want to fix this straight away, you could fork the vcpkg tree and add the `set(VCPKG_BUILD_TYPE release)` line to the required triplets and change the `url=` and `branch=` in `Cargo.toml`.

That said, I'll try and get some way to do it out shortly.

**Comment by pkgw** [comment]
Yeah, things seem to be under control in the Tectonic CI for now so I see no need to fork, but I'm definitely interested to test out any potential fixes.

(That being said, the vcpkg builds seem to be taking longer and longer ... seems like the msys servers are getting slower or something.)

The other thing specific to Tectonic is that several of the deps, including some slow-to-build ones like iconv, are coming from `fontconfig`, and I'm not sure if we even need that on Windows. Now that Tectonic's multi-crate organization is coming together I'm in a better position to experiment on that front.
